p2p/peer.py

```diff
@@ -481,6 +484,15 @@ def process_msg(self, cmd: protocol.Command, msg: protocol.PayloadType) -> None:
             await self.disconnect(DisconnectReason.bad_protocol)
             raise HandshakeFailure(f"Expected a Hello msg, got {cmd}, disconnecting")
 
+        # limit number of chars to be displayed, and try to keep printable ones only
+        # MAGIC 256: arbitrary, "should be enough for everybody"
+        original_version = msg['client_version_string']
+        client_version_string = original_version[:256] + ('...' if original_version[256:] else '')
+        if client_version_string.isprintable():
+            self.client_version_string = client_version_string.strip('\t\n\r\x0b\x0c')
```

Comment:
Strip is a noop here, though, because:

```py
In [3]: for char in '\t\n\r\x0b\x0c':
   ...:     print(repr(char), char.isprintable())    
   ...:     
'\t' False
'\n' False
'\r' False
'\x0b' False
'\x0c' False
```

If you like, we could do `.strip()` to get rid of useless whitespace on the edges. Think I'd rather not leave the noop strip in, though.

```suggestion
            self.client_version_string = client_version_string.strip()
```

Reply:
Argh! My idea with this was to prevent

```
   DEBUG  03-06 14:07:26           ETHPeerPool  ETHPeer <Node(0x04f8@88.198.47.121)>: uptime=0:00:41:29, received_msgs=1076, most_received=Transactions (cmd_id=18)(576)                                                                      
   DEBUG  03-06 14:07:26           ETHPeerPool  client_version_string='Geth/
╔════════════════════╗
║    BUY SCAMCOIN    ║
╚════════════════════╝
v1.8.11-stable-dea1ce05/linux-amd64/go1.10'      
   DEBUG  03-06 14:07:26           ETHPeerPool      BlockBodies: None                                                                                                                                                                         
   DEBUG  03-06 14:07:26           ETHPeerPool      BlockHeaders: msgs=1  items=2  rtt=19.01/0.19/0.00  ips=0.51390  timeouts=0  quality=5                                                                                                    
   DEBUG  03-06 14:07:26           ETHPeerPool      NodeData: None                                                                                                                                                                            
   DEBUG  03-06 14:07:26           ETHPeerPool      Receipts: msgs=283  items=1523595  rtt=1.14/1.93/0.21  ips=5241.20800  timeouts=32  quality=10
```

but I failed to see that the characters are non-printable (and would therefore force processing the string with `repr()`).

One sec...